shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/Generator/Utxo.hs

```diff
@@ -120,18 +124,47 @@ genTx ::
   LedgerEnv ->
   (UTxOState era, DPState era) ->
   Gen (Tx era)
-genTx ge@(GenEnv _ (Constants {genTxRetries})) =
-  genTxRetry genTxRetries ge
+genTx ge le st =
+  tryGenTx ge le st >>= \case
+    Left exn -> Exn.throw exn
+    Right tx -> pure tx
+
+data GenTxException
```

Comment:
It's called "-Exception", but it isn't an instance of the `Exception` class and thus never actually "thrown". I typically use `-Error` for that.

Reply:
"Alternative exit status"

Reply:
It is an instance (see just below) and it is thrown (see just above). Or am I confused about your observation.